### PR TITLE
Add VB compile build events

### PR DESCRIPTION
**Pull request by basoundr**
This is a follow up PR for #2004. Only the last commit is the new commit and the rest of the commits are part of #2004 and #2005.

**Customer scenario**

Enable VB Compile page - Build events page.

**Bugs this fixes:** 

#346 

**Workarounds, if any**

None

**Risk**

Low risk - since this just affects that page

**Performance impact**

None - since this is a new page, which just reflects the msbuild properties like any other page.

**Is this a regression from a previous update?** N/A

**Root cause analysis:** N/A

**How was the bug found?** N/A

### CODE REVIEW

**Comment by basoundr** [comment]
Tagging @davkean @srivatsn @dotnet/project-system for review

File: src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Properties/OptionCompareEnumProvider.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ExportDynamicEnumValuesProvider("OptionCompareEnumProvider")]
+    [AppliesTo(ProjectCapability.VisualBasic)]
+    internal class OptionCompareEnumProvider : IDynamicEnumValuesProvider
+    {
+        private readonly Dictionary<string, IEnumValue> _listedOptionCompareEnumValues = new Dictionary<string, IEnumValue>
+            {
+                { "Binary", new PageEnumValue(new EnumValue {Name = "Binary",   DisplayName = "Binary", IsDefault = true }) },
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Properties/OptionCompareEnumProvider.cs]: I don't understand this - if you are not mapping between values, then why does this need to be a DynamicEnumProperty? Why not just EnumProperty and list the values as Binary and Text?

**Comment by basoundr** [comment]
@srivatsn Addressed your feedback

**Comment by basoundr** [comment]
Tagging @MattGertz for approval

**Pull request merged**